registry/infrastructure/repositories/service_publisher_repository.py

```diff
@@ -185,3 +185,15 @@ def get_list_of_service_pending_for_approval(self, limit):
             self.session.rollback()
             raise e
         return services
+
+    def get_service_state_with_status(self, status):
+        service_state_db = self.session.query(ServiceState).filter(ServiceState.state == status).all()
+        service_state = ServiceFactory.convert_service_state_from_db_list(service_state_db)
+        self.session.commit()
+        return service_state
+
+    def update_service_status(self, service_uuid_list, prev_state, next_state):
+        self.session.query(ServiceState).filter(ServiceState.service_uuid.in_(service_uuid_list))\
+            .filter(ServiceState.state == prev_state).update({ServiceState.state: next_state},
+                                                             synchronize_session='fetch')
```

Comment:
Passing the fetch argument will basically run the query twice, once as a select and once as an update.
If running two queries is not desired, pass synchronize_session=False instead and then call session.expire_all() immediately after the delete to avoid having inconsistent state within the MetaData store.
https://docs.sqlalchemy.org/en/13/orm/query.html#sqlalchemy.orm.query.Query.delete